rust/threshold-ecdsa/src/ecdsa_example_rust/src/lib.rs

```diff
@@ -19,82 +22,38 @@ struct SignatureVerificationReply {
     pub is_signature_valid: bool,
 }
 
-type CanisterId = Principal;
-
-#[derive(CandidType, Serialize, Debug)]
-struct ECDSAPublicKey {
-    pub canister_id: Option<CanisterId>,
-    pub derivation_path: Vec<Vec<u8>>,
-    pub key_id: EcdsaKeyId,
-}
-
-#[derive(CandidType, Deserialize, Debug)]
-struct ECDSAPublicKeyReply {
-    pub public_key: Vec<u8>,
-    pub chain_code: Vec<u8>,
-}
-
-#[derive(CandidType, Serialize, Debug)]
-struct SignWithECDSA {
-    pub message_hash: Vec<u8>,
-    pub derivation_path: Vec<Vec<u8>>,
-    pub key_id: EcdsaKeyId,
-}
-
-#[derive(CandidType, Deserialize, Debug)]
-struct SignWithECDSAReply {
-    pub signature: Vec<u8>,
-}
-
-#[derive(CandidType, Serialize, Debug, Clone)]
-struct EcdsaKeyId {
-    pub curve: EcdsaCurve,
-    pub name: String,
-}
-
-#[derive(CandidType, Serialize, Debug, Clone)]
-pub enum EcdsaCurve {
-    #[serde(rename = "secp256k1")]
-    Secp256k1,
-}
 
 #[update]
 async fn public_key() -> Result<PublicKeyReply, String> {
-    let request = ECDSAPublicKey {
+    let request = EcdsaPublicKeyArgument {
         canister_id: None,
         derivation_path: vec![],
         key_id: EcdsaKeyIds::TestKeyLocalDevelopment.to_key_id(),
     };
 
-    let (res,): (ECDSAPublicKeyReply,) =
-        ic_cdk::call(mgmt_canister_id(), "ecdsa_public_key", (request,))
-            .await
-            .map_err(|e| format!("ecdsa_public_key failed {}", e.1))?;
+    let (response,) = ecdsa_public_key(request)
+        .await
+        .map_err(|e| format!("ecdsa_public_key failed {}", e.1))?;
 
     Ok(PublicKeyReply {
-        public_key_hex: hex::encode(&res.public_key),
+        public_key_hex: hex::encode(response.public_key),
     })
 }
 
 #[update]
 async fn sign(message: String) -> Result<SignatureReply, String> {
-    let request = SignWithECDSA {
+    let request = SignWithEcdsaArgument {
         message_hash: sha256(&message).to_vec(),
         derivation_path: vec![],
         key_id: EcdsaKeyIds::TestKeyLocalDevelopment.to_key_id(),
     };
 
-    let (response,): (SignWithECDSAReply,) = ic_cdk::api::call::call_with_payment(
-        mgmt_canister_id(),
-        "sign_with_ecdsa",
-        (request,),
-        25_000_000_000,
-    )
-    .await
-    .map_err(|e| format!("sign_with_ecdsa failed {}", e.1))?;
+    let (response,) = sign_with_ecdsa(request)
```

Comment:
Somewhat out of scopehere, but `sign_with_ecdsa` handles the cycles cost under the hood, and it would be great if the [SIGN_WITH_ECDSA_FEE constant](https://docs.rs/ic-cdk/0.14.0/src/ic_cdk/api/management_canister/ecdsa/mod.rs.html#9) that is [used](https://docs.rs/ic-cdk/0.14.0/src/ic_cdk/api/management_canister/ecdsa/mod.rs.html#32) internally was public, so that developers can notice if the fee changes (cc @sesi200).